go-controller/pkg/ovn/namespace.go

```diff
@@ -240,6 +245,41 @@ func (oc *Controller) AddNamespace(ns *kapi.Namespace) {
 
 	// If multicast enabled, adds all current pods in the namespace to the allow policy
 	oc.multicastUpdateNamespace(ns, nsInfo)
+
+	// ensure any missing network policies are added
+	missingPolicies := make(map[*v1.NetworkPolicy]*networkPolicy)
+
+	for _, policy := range netPolicies {
+		if _, alreadyExists := nsInfo.networkPolicies[policy.Name]; !alreadyExists {
+			if len(nsInfo.networkPolicies) == 0 {
+				if err := oc.ensureDefaultDenyPortGroup(nsInfo, ns.Name, policy.Name); err != nil {
+					klog.Error(err.Error())
+					nsInfo.Unlock()
+					return
+				}
+			}
+
+			np := NewNetworkPolicy(policy)
+			// lock the np to prevent a update/delete event in policy handler from getting it
+			// as we must create it first
+			np.Lock()
```

Comment:
@trozet where do we *un*lock the NP?

Reply:
Seems like createNetworkPolicy().... but that should be better documented for sure both here and for createNetworkPolicy.

Reply:
i'm moving it to into the same function where it gets unlocked on the next revision